src/tls_relation.py

```diff
@@ -207,16 +209,22 @@ def certificate_relation_created(  # type: ignore[no-untyped-def]
             "key": private_key.decode(),
         }
         if JujuVersion.from_environ().has_secrets:
+            secret_id = ""  # nosec
             try:
                 secret = charm.model.get_secret(label=f"private-key-{hostname}")
                 secret.set_content(private_key_dict)
+                secret_id = secret.id
             except SecretNotFoundError:
                 secret = charm.app.add_secret(
                     content=private_key_dict, label=f"private-key-{hostname}"
                 )
-        self.update_relation_data_fields(private_key_dict, tls_certificates_relation, charm.app)
-        peer_relation = charm.model.get_relation("nginx-peers")
-        self.update_relation_data_fields(private_key_dict, peer_relation, charm.app)
+                secret_id = secret.id
```

Comment:
You probably don't need to repeat this line twice and directly use `secret.id` later on